pkg/internal/util/kubeconfig/write_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"sigs.k8s.io/kind/pkg/internal/util/assert"
+)
+
+func TestWrite(t *testing.T) {
+	t.Parallel()
+	t.Run("non-existent file", testWriteNoExistingFile)
```

Comment:
inline the function here?
i don't prefer lambdas or closures, but it seems to be the testing pattern in go.

Reply:
Shrug, IMO this pattern is nicer for large test methods so you can more easily read what the cases are more quickly 😅
Functionally it's the same so.